### ISSUE TITLE
[rush] Add --verbose flag to rushx cut down on current line-noise

### ISSUE THREAD

**Issue by dustinlacewell** [microsoft/rushstack#2815]
## Summary

`rushx` currently outputs a good amount of text when you run anything with it. When trying to run CLI projects this can be a bit distracting. `rushx` should have a `--verbose` flag behind which these current lines of output would be hidden.

Example,

```
rushx start hosts disable bar
Found configuration in /home/me/ext/dnsbot2/rush.json


Rush Multi-Project Build Tool 5.47.0 - https://rushjs.io
Node.js version is 14.17.3 (LTS)

Found configuration in /home/me/ext/dnsbot2/rush.json

Executing: "node --no-warnings dist/index.js hosts disable bar"

Updating all records...
```
Only the last line is output from my actual project.

## Repro steps

  **Expected result:** 

  **Actual result:** 

## Standard questions

Please answer these questions to help us investigate your issue more quickly:

| Question | Answer |
| -------- | -------- |
| `@microsoft/rush` globally installed version? | 5.48.0 |
| `rushVersion` from rush.json? | 5.47.0 |
| `useWorkspaces` from rush.json? | false |
| Operating system? | linux |
| Node.js version (`node -v`)? | 14.17.3 |

**Comment by iclanton** [comment]
Yeah that's a good idea. `rushx`'s handling of command-line arguments is currently a little lacking. @octogonz, I remember you wanting to fix some of these issues.

**Comment by octogonz** [comment]
> 
> 
> Yeah that's a good idea. `rushx`'s handling of command-line arguments is currently a little lacking. @octogonz, I remember you wanting to fix some of these issues.

Last week @elliot-nelson and I put together a dev plan, and I started work on a prototype for an overhaul of the `ts-command-line` parser.  We should have an update soonish.

**Comment by elliot-nelson** [comment]
Although a bigger overhaul is pending, we probably don't need to block this improvement on that -- I put up a strawman with one possible approach.
